docs/understanding-tracking-design/json-schema-type-casting-rules/index.md

```diff
@@ -0,0 +1,370 @@
+---
+title: "Type casting rules for json schemas"
+date: "2021-12-30"
+sidebar_position: 100
+hide_table_of_contents: true
+---
+
+```mdx-code-block
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+```
+
+_The row order in this table is important.  Type lookup stops after first match is found scanning from top to bottom (with the single exception of "null" - first row in the table)_
+<Tabs groupId="type-casting">
+  <TabItem value="redshift" label="Redshift" default>
+
+<table>
+<thead>
+<td>Json Schema</td>
+<td>Redshift Type</td>
+<td>Notes</td>
+</thead>
+<tbody>
+<tr>
+<td >
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": ["null", T1, ...]<br/>
+&#125;</pre>
+OR
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"enum": ["null", E1, ...]<br/>
+&#125;</pre>
+</td>
+<td >
+<code>NULLABLE</code>
+</td>
+<td><code>"null"</code> is not considered for type casting logic. Only for nullability constraint. Type lookup will continue down the table.
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"enum": [E1, E2, ...]<br/>
+&#125;</pre>
+</td>
+<td><code>VARCHAR(M)</code></td>
+<td> Where <code>M</code> is maximum size of <code>json.stringify(E*)</code><br/><br/>
+<code>enum</code> contains more than <b>one</b> JavaScript type - <code>string</code>, <code>number|integer</code>, <code>boolean</code>.<br/>
+For the purposes of this  <code>number</code> and <code>integer</code> are the same.<br/>
+<br/>
+<code>array</code>, <code>object</code>, <code>NaN</code> and other types in enum will error.   
```

Comment:
```suggestion
<code>array</code>, <code>object</code>, <code>NaN</code> and other types in enum will result in an error.
```
Error where? Iglu? Failed events of “invalid schema” type? Failed events in the transformer? Something else?

Reply:
the will fallback to the last row.